x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_logic.ts

```diff
@@ -187,4 +220,123 @@ export const ResultSettingsLogic = kea<MakeLogicType<ResultSettingsValues, Resul
       },
     ],
   }),
+  selectors: ({ selectors }) => ({
+    resultFieldsAtDefaultSettings: [
+      () => [selectors.resultFields],
+      (resultFields: FieldResultSettingObject) => areFieldsAtDefaultSettings(resultFields),
+    ],
+    resultFieldsEmpty: [
+      () => [selectors.resultFields],
+      (resultFields: FieldResultSettingObject) => areFieldsEmpty(resultFields),
+    ],
+    stagedUpdates: [
+      () => [selectors.lastSavedResultFields, selectors.resultFields],
+      (lastSavedResultFields: FieldResultSettingObject, resultFields: FieldResultSettingObject) =>
+        !isEqual(lastSavedResultFields, resultFields),
+    ],
+    reducedServerResultFields: [
+      () => [selectors.serverResultFields],
+      (serverResultFields: ServerFieldResultSettingObject) =>
+        Object.entries(serverResultFields).reduce(
+          (acc: ServerFieldResultSettingObject, [fieldName, resultSetting]) => {
+            if (resultSetting.raw || resultSetting.snippet) {
+              acc[fieldName] = resultSetting;
+            }
+            return acc;
+          },
+          {}
+        ),
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    clearRawSizeForField: (fieldName) => {
+      actions.updateField(fieldName, omit(values.resultFields[fieldName], ['rawSize']));
+    },
+    clearSnippetSizeForField: (fieldName) => {
+      actions.updateField(fieldName, omit(values.resultFields[fieldName], ['snippetSize']));
+    },
+    toggleRawForField: (fieldName: string) => {
+      // We cast this b/c it could be an empty object, which we can still treat as a FieldResultSetting safely
```

Comment:
I'm not a huge fan of stuff like "b/c" in comments because it can be confusing to non-native English speakers.

Reply:
https://github.com/elastic/kibana/pull/94947/commits/aa858dfea52082d6a879a69b1a76e94eed732b56